cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/TableOutputFormatter.cs

```diff
@@ -0,0 +1,168 @@
+using System.Text.Json;
+using Spectre.Console;
+using Spectre.Console.Rendering;
+
+namespace Microsoft.Kiota.Cli.Commons.IO;
+
+/// <summary>
+/// The table output formatter
+/// </summary>
+public class TableOutputFormatter : IOutputFormatter
+{
+    private readonly IAnsiConsole _ansiConsole;
+
+    /// <summary>
+    /// Creates a new table output formatter with a default console
+    /// </summary>
+    public TableOutputFormatter() : this(AnsiConsole.Console)
+    {
+    }
+
+    /// <summary>
+    /// Creates a new table output formatter with the provided console
+    /// </summary>
+    /// <param name="console">The console to use</param>
+    public TableOutputFormatter(IAnsiConsole console)
+    {
+        _ansiConsole = console;
+    }
+
+    /// <inheritdoc />
+    public async Task WriteOutputAsync(Stream content, IOutputFormatterOptions options, CancellationToken cancellationToken = default) {
+        using var doc = await JsonDocument.ParseAsync(content, cancellationToken: cancellationToken);
+        var table = ConstructTable(doc);
+        _ansiConsole.Write(table);
+    }
+
+    /// <summary>
+    /// Construct a table given a JSON document
+    /// </summary>
+    /// <param name="document">The parsed json document</param>
+    /// <returns>A table</returns>
+    public Table ConstructTable(JsonDocument document) {
```

Comment:
switch back to internal, and use the internal visible to assembly attribute so people don't take a dependency on that.